paperqa/agents/tools.py

```diff
@@ -373,6 +370,10 @@ async def complete(
         # TODO: eliminate race condition here if agent calls 2+ times in parallel
         # with opposite has_successful_answer values
         state.session.has_successful_answer = has_successful_answer
+
+        if not state.session.answer:
```

Comment:
Can you make test assertions for this in `test_agent_sharing_state`? You can add a `complete` tool call right at the start of the test